src/controllers/v1/rolePermissionMapping.js

```diff
@@ -44,4 +44,21 @@ module.exports = class RolePermission {
 			return error
 		}
 	}
+
+	/**
+	 * list rolePermission.
+	 * @method
+	 * @name list
+	 * @param {Integer} req.decodedToken.roles - roles
+	 * @returns {JSON} - RolePermission list object.
+	 */
+
+	async list(req) {
+		try {
+			const roleIds = Array.isArray(req.decodedToken.roles) ? req.decodedToken.roles.map(({ id }) => id) : []
```

Comment:
You are doing an array check here. First verify if this roles will ever be anything other than an array. If it is always going to be an array, then you can remove this check and then just do a map on req.decodedToken.roles.

Reply:
I verified it will work without array check , it comes in array format only.